Automattic/wp-calypso
# refactor celebration modals to expose their logic

Broken out from https://github.com/Automattic/wp-calypso/pull/76267.

In https://github.com/Automattic/wp-calypso/pull/76267, I'm adding a new sharing modal which will appear after publishing a post. However it is important that that modal does not overwrite these existing modals.

This PR prepares the way by exposing the logic of whether the videopress or seller modal should be shown into `use-*` hooks. In https://github.com/Automattic/wp-calypso/pull/76267, I can then add logic to only display the new modal if neither of these existing modals are going to be shown.

### Testing instructions 
Check out this branch, build the ETK plugin and upload it to your sandbox with:
```
cd apps/editing-toolkit
yarn dev --sync
```

Note: When I tested both modals saving in the Site Editor, I found that they were not shown. In both cases, the isEditorSaving flag was not updated on save. This appears to be an existing bug, and I'll leave this out of this PR (although, it's possible I'm not using the site editor in the way that is expected).

#### Video celebration modal:
* **Sandbox** a site with `site_intent` set to `videopress`
  * in `wpsh` run `update_blog_option( $site_id, 'site_intent', 'videopress' )`
  * in `wpsh` run `update_blog_option( $site_id, 'has_seen_video_celebration_modal', false )`
* Create a post and add a video block to it.
* Publish the post
* You should see the videopress celebration modal.

<img width="666" alt="Screen Shot 2023-05-04 at 2 52 51 pm" src="https://user-images.githubusercontent.com/22446385/236115225-305a6e01-ba25-47e8-a2c5-d5b2fe931ce1.png">

Note: the [launchpad modal](https://github.com/Automattic/jetpack/blob/f60b8721717189e53129e856c75a9461a0d153fb/projects/plugins/jetpack/extensions/plugins/launchpad-save-modal/launchpad-save-modal.php) seems to already be conflicting with the videopress celebration modal to an extent, but they appear behind one another


#### Seller celebration modal:
* **Sandbox** a site with `site_intent` set to `sell`
  * in `wpsh` run `update_blog_option( $site_id, 'site_intent', 'sell' )`
  * in `wpsh` run `update_blog_option( $site_id, 'has_seen_video_celebration_modal', false )`
* In the post editor, add a "payments buttons" block,
* Save the page
* You should see the seller celebration modal
<img width="944" alt="Screen Shot 2023-05-04 at 12 47 16 pm" src="https://user-images.githubusercontent.com/22446385/236100285-a10d4653-bae7-4835-af63-44d27ca9d839.png">

#### First post published modal:
* **Sandbox** a site with `site_intent` set to `sell`
  * in `wpsh` run `update_blog_option( $site_id, 'site_intent', 'write' )`
  * in `wpsh` run `update_blog_option( $site_id, 'has_never_published_post', true )`
* In the post editor, publish a post.
* You should see the first post published modal

<img width="649" alt="Screen Shot 2023-05-05 at 12 54 09 pm" src="https://user-images.githubusercontent.com/22446385/236368596-25221a3b-b779-4f9d-9aca-e805467b0a6d.png">
